File: core/baker-interface/src/main/scala/com/ing/baker/runtime/inmemory/InMemoryInteractionManager.scala
package com.ing.baker.runtime.inmemory

import cats.effect.IO
import com.ing.baker.runtime.model.{InteractionInstance, InteractionManager}

object InMemoryInteractionManager {

  def build(implementations: List[InteractionInstance[IO]]): IO[InMemoryInteractionManager] =
    IO(new InMemoryInteractionManager(implementations))
}

final class InMemoryInteractionManager(implementations: List[InteractionInstance[IO]]) extends InteractionManager[IO] {

  override val allowSupersetForOutputTypes: Boolean = false

  override def listAll: IO[List[InteractionInstance[IO]]] = IO(implementations)

}

File: core/baker-runtime-in-memory/src/main/kotlin/com/ing/baker/runtime/inmemory/InMemoryInteractionManager.kt
package com.ing.baker.runtime.inmemory

import cats.effect.IO
import com.ing.baker.runtime.model.InteractionInstance
import com.ing.baker.runtime.model.InteractionManager
import scala.collection.immutable.List as ScalaList

class InMemoryInteractionManager(private val implementations: ScalaList<InteractionInstance<IO<*>>>): InteractionManager<IO<*>> {

    override fun allowSupersetForOutputTypes(): Boolean = false

    override fun listAll(): IO<ScalaList<InteractionInstance<IO<*>>>> = IO.pure(implementations)
}
